src/router/storeSync/storeSync.routerPlugin.js

```diff
@@ -191,9 +199,39 @@ function urlQueryWatcher(store, to) {
  */
 const storeSyncRouterPlugin = (router, store) => {
     let unsubscribeStoreMutation = null
-    router.beforeEach((to) => {
+    router.beforeEach((to, from) => {
+        log.debug('[Router store plugin] Entering the store sync plugin')
+        // we define a return Value, so we can check across the function what its value is
+        let retVal = undefined
+        log.debug('[Router store plugin] We received the following `from` :', from)
+        log.debug('[Router store plugin] We expect to go to the following `to` :', to)
+
+        if (!MAP_VIEWS.includes(to.name)) {
+            // leaving MapView make sure to unsubscribe the store mutation
+            if (unsubscribeStoreMutation) {
+                log.info(
+                    `[Router store plugin] Leaving ${to.name}, unregister store mutation watcher`
+                )
+
+                unsubscribeStoreMutation()
+                retVal = undefined
+            }
+        }
+        if (MAP_VIEWS.includes(to.name) && store.state.app.isReady) {
+            // Synchronize the store with the url query only on MapView and when the application
+            // is ready
+            retVal = urlQueryWatcher(store, to, from)
+        }
+
+        log.debug('[Router store plugin] exiting navigation guard with the following value', retVal)
+        // Note we return undefined to validate the route, see Vue Router documentation
+        return retVal
+    })
+    router.afterEach((to, from) => {
         if (MAP_VIEWS.includes(to.name) && !unsubscribeStoreMutation) {
```

Comment:
A blank line before and a comment about why we need to do this afterEach would be good.